docs/performance-guide.md

```diff
@@ -180,95 +264,122 @@ group.
+| Group count               | 10    | 10    | 10     | 10     | 10     | 10      | 10
+| Inbound (message/s)  | 40    | 20    | 20     | 10     | 20     | 20      | 20      |
+| Inbound Bandwidth (byte/s)  | 80K   | 40K   | 40K    | 20K    | 40K    | 40K     | 40K     |
+| Outbound (message/s) | 4,000 | 4,000 | 10,000 | 10,000 | 40,000 | 100,000 | 200,000 |
+| Outbound Bandwidth (byte/s) | 8M    | 8M    | 20M    | 20M    | 80M    | 200M    | 400M    |
 
-App server number suggested
+The sending connection count is no more than 40, then the burden on app server is small, thus the suggested web app number is also small.
 
 |                  | Unit1 | Unit2 | Unit5 | Unit10 | Unit20 | Unit50 | Unit100 |
 |------------------|-------|-------|-------|--------|--------|--------|---------|
 | Connections      | 1,000 | 2,000 | 5,000 | 10,000 | 20,000 | 50,000 | 100,000 |
 | App server count | 2     | 2     | 2     | 2      | 2      | 2      | 2       |
 
+> **Note**
+>
+> Increase the default server connections from 5 to 40 on every app server to
+> avoid possible unbalanced server connections to ASRS.
+> 
+> The client connection number, message size, message sending rate, routing cost, and unit tier take impact on overall performance of **send to big group**.
 
 <a name="sendToClient"></a>
 ### Send to connection
 
-Clients get their own connection ID before starting sending message. The
-performance benchmark is responsible to collect all connection IDs, shuffle them and re-assign them to all clients as a sending target. Every client gets a
-target connection ID, and the clients will always send message to the target connection.
+In this use case, when clients establish the connections to ASRS, every client calls a special hub to get their own connection ID. The performance benchmark is responsible to collect all connection IDs, shuffle them and re-assign them to all clients as a sending target. The clients keep on sending message to the target connection until the performance test finishes.
 
 ![Send to client](./images/sendtoclient.png)
 
+The routing cost for **Send to connection** is a little different from the others.
+
+For one ASRS instance, there are at least 2 runtime instances for SLA consideration. So, the sending connection and target connection may or may not belong to the same runtime instance. If they are in the same runtime instance, the routing cost is relative low because the message is sent to a *local* connection. But if they cross runtime instance, the routing cost is high for the message is sent to a *global* connection which requires the help of the distributed data structure.
+
+More runtime instances, more possibility to see sending and target connections across runtime instances. As the proportion of sending message to *global* connection increases, the overall performance is limited by routing cost. Unit 50 has reached the limit. As a result, unit 100 cannot improve further.
```

Comment:
This explanation has too many implementation details, I think it's better to just use the "routing cost" to explain (like for small groups).

Reply:
All right